rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/CoordinatorBookKeeper.java

```diff
@@ -84,6 +87,43 @@ public void handleHeartbeat(String workerHostname, HeartbeatStatus heartbeatStat
     }
   }
 
+  @Override
+  public List<String> getNodeHostNames(int clusterType)
```

Comment:
Why do we need this in interface? This could be a private method inside CoordBookKeeper

Reply:
We need this because worker bookkeeper invokes this method to populate its local cache.
https://github.com/qubole/rubix/pull/194/files#diff-85292be7f2f40e0c26f9f055cbadbcb0R79

Reply:
The logic in getClusterNodeHostName is 
1. get the list of nodes.
2. invoke cluster manager getNodeIndex with number of nodes and the file path to get the nodeIndex
3. Use nodeIndex to get the actual hostname from the list of nodes.

The difference between Coordinator and Worker implementation is at step 1 (how the list of nodes is fetched)
We can extract step 2 and 3  to a common method. Thats only two lines of code. 
The main logic of calculating the node index and getting the list of nodes are already pushed into cluster manager classes.